packages/amplify_datastore/example/integration_test/auth_test/private_blog_allow_read.dart

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+import 'dart:math';
+
+import 'package:amplify_api/amplify_api.dart';
+import 'package:amplify_auth_cognito/amplify_auth_cognito.dart';
+import 'package:amplify_datastore_example/auth_models/private_blog_allow_read/ModelProvider.dart';
+import 'package:amplify_test/amplify_test.dart';
+import 'package:integration_test/integration_test.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:amplify_flutter/amplify_flutter.dart';
+
+import '../utils/setup_utils.dart';
+import '../utils/test_users.dart';
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  Random random = Random();
+
+  // Create TestUsers that can be utilized for entire test run
+  TestUser user1 = TestUser<Blog>(generateUsername(), generatePassword(), []);
+  TestUser user2 = TestUser<Blog>(generateUsername(), generatePassword(), []);
+
+  String _generateBlogName() {
+    return random.nextInt(500).toString();
+  }
+
+  /// Creates a blog as [user1], saves it, and signs in [user2]
+  Future<void> _blogSetupAndTest(String blogName) async {
+    // Sign in with user1 and start the store
+    await user1.signInCognitoUser();
+
+    // Create blog as user1
+    await user1.createModels(Blog(name: blogName));
```

Comment:
Can this redundant code be refactored into a helper method that can be called from different auth_test s?

Reply:
In future, maybe we can investigate a concise way to represent unit/integration tests

Appears most tests share a lot of common setup/teardown and inner testing logic.  But long code file makes it harder to read/parse/maintain them all. 

It would be interesting if we could have our own inner symbols to represent maybe : 

```
blogSetupAndTest() async {
    runIntegrationTest( new List<IntegrationTestSteps>() {
SignIn(User1), CreateAndVerifyModel(User1, Blog(blogName)), Signout(User1), Signout(User2) 
}); 
}
```

Or maybe represent it as a txt file we read in and parse. 

Such a representation could make it easier to make sure we capture each permutation that might cause an error, and if well represented, allow for some rudimentary auto generation of integration tests.